### ISSUE TITLE
Beta and RC tags not working as expected

### ISSUE THREAD

**Issue by phw** [brlin-tw/selective-checkout#4]
I tried to use the configuration for beta and RC tags, but this seems not to work. This is my call to selective-checkout:

```
"$SNAPCRAFT_STAGE"/scriptlets/selective-checkout \
    --release-tag-prefix
    --release-tag-prefix="release-" \
    --stable-tag-pattern="[[:digit:]]+\.[[:digit:]]+(\.[[:digit:]]+)?$" \
    --beta-tag-pattern="[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+b[[:digit:]]+$" \
    --release-candidate-tag-pattern="[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+rc[[:digit:]]+$"
```

This is for https://github.com/metabrainz/picard . The tags there are like:

stable: release-2.5.6
rc: release-2.3.0rc1
beta: release-2.6.0b1

Now both RC and beta don't behave as I think they should.

**RC:**

selective-checkout found the old tag for 2.3.0rc1 and wants to build this, but the current version published in the RC channel is 2.5.6, which is greater than 2.3.0rc1. So it should actually ignore this tag.

Expected behavior: Do not try to build and publish the RC tag, if it is older than the version currently published in the RC channel. 

**Beta:**

For beta it does not try to build the release-2.6.0b1 tag, even though this is newer than the 2.5.6 currently in the beta channel. Instead a normal dev build is triggered.

Expected behavior: If there is a beta tag and it is newer than what is currently in the beta channel, build and publish this tag

**Comment by phw** [comment]
The snapcraft.yml for this is at https://github.com/phw/musicbrainz-picard-snap/blob/master/snap/snapcraft.yaml

RC  build is currently disabled because it misbehaves as described above. But would like to get the beta build running.

**Comment by Lin-Buo-Ren** [comment]
Thanks for the report, I'll check it out.

**Comment by brlin-tw** [comment]
Hello, apologies for the ignorance.

We believe this bug is being fixed in 3.0.0, you may refer to the checkout determination logic [here](https://github.com/brlin-tw/selective-checkout/blob/master/scriptlets/selective-checkout#L983-L1154), please follow up if it isn't the case, thanks!
